### ISSUE TITLE
Support f-strings in `Annotated` type annotations when using an alias of `Annotated`

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#6714]
This is related to [this discussion](https://github.com/microsoft/pyright/discussions/6713#discussioncomment-7827524).

```python
import numpy as np
from jaxtyping import Float

def f(n):
    # pyright complains here
    def g(x: Float[np.ndarray, f"m {n}"]) -> Float[np.ndarray, f"{n}"]:
        return x.sum(axis=0)
    return g

print(f(4)(np.ones((2, 4))))
```

**Comment by erictraut** [comment]
This will be included in the next release.

In addition to supporting f-strings, I also addressed the code paths for the unpack operator, assignment expressions (walrus operator), the `await` operator, chained strings, and strings with escape sequences. All of these are now allowed in an aliased `Annotated` type annotation.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.340, which I just published. It will be included in a future release of pylance.
